src/Cairo.jl

```diff
@@ -311,10 +323,10 @@ end
 
 function write_to_png(surface::CairoSurface, filename::AbstractString)
     ccall((:cairo_surface_write_to_png,_jl_libcairo), Void,
-          (Ptr{UInt8},Ptr{UInt8}), surface.ptr, bytestring(filename))
+          (Ptr{UInt8},Ptr{UInt8}), surface.ptr, @compat(String(filename)))
 end
 
-writemime(io::IO, ::MIME"image/png", surface::CairoSurface) =
+@compat Base.show(io::IO, ::MIME"image/png", surface::CairoSurface) =
```

Comment:
We can amend metadata if you'd like to be especially thorough, sure. Minimum versions not being recent enough hasn't caused too many issues yet (adding an upper bound on Compat would be a bit strange), but I think it's worth generally keeping in mind.